.github/workflows/downstream-project-spl.yml

```
@@ -0,0 +1,103 @@
+name: Downstream Project - SPL
+
+on:
+  push:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+
+  pull_request:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+
+    paths:
+      - "**.rs"
+      - "Cargo.toml"
+      - "Cargo.lock"
+      - "cargo-build-bpf"
+      - "cargo-test-bpf"
+      - "cargo-build-sbf"
+      - "cargo-test-sbf"
+      - "ci/downstream-projects/run-spl.sh"
+      - ".github/workflows/downstream-project-spl.yml"
+  workflow_call:
+    inputs:
+      branch:
+        required: false
+        type: string
+        default: "master"
+
+env:
+  SHELL: /bin/bash
+  SCCACHE_GHA_ENABLED: "true"
+  RUSTC_WRAPPER: "sccache"
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - shell: bash
+        run: |
+          .github/scripts/purge-ubuntu-runner.sh
+
+      - uses: mozilla-actions/sccache-action@v0.0.3
+
+      - shell: bash
+        run: |
+          source .github/scripts/downstream-project-spl-common.sh
+
+          sudo apt update
+          sudo apt install libudev-dev binutils-dev libunwind-dev -y
+
+          $CARGO_BUILD_SBF --manifest-path token/program/Cargo.toml
+          $CARGO_BUILD_SBF --manifest-path token/program-2022/Cargo.toml
+          $CARGO_BUILD_SBF --manifest-path token-upgrade/program/Cargo.toml
+          $CARGO_BUILD_SBF --manifest-path associated-token-account/program/Cargo.toml
+          $CARGO_BUILD_SBF --manifest-path stake-pool/single-pool/Cargo.toml
+
+          cargo build
+          cargo test
+
+  cargo-test-sbf:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        programs:
+          - [token/program]
+          - [
+              instruction-padding/program,
+              token/program-2022,
+              token/program-2022-test,
+            ]
+          - [
+              associated-token-account/program,
+              associated-token-account/program-test,
+            ]
+          - [token-upgrade/program]
+          - [feature-proposal/program]
+          - [governance/addin-mock/program, governance/program]
+          - [memo/program]
+          - [name-service/program]
+          - [stake-pool/program]
+          - [stake-pool/single-pool]
```

https://github.com/solana-labs/solana-program-library/pull/5010 was just merged so:
```suggestion
          - [single-pool/program]
```